subsys/bluetooth/host/conn.c

```diff
@@ -1205,28 +1207,40 @@ void bt_conn_set_state(struct bt_conn *conn, bt_conn_state_t state)
 	}
 }
 
-struct bt_conn *bt_conn_lookup_handle(uint16_t handle)
+struct bt_conn *bt_conn_lookup_handle(uint16_t handle, int type)
```

Comment:
This feels like the wrong way of doing this IMO. 

`bt_conn_lookup_handle` is a local function to get a local `bt_conn` based on a local handle. By "local" I mean this isn't related to the controller, which is what we seem to want to guard against. 

IMO any caller of this function that want to verify the type, should do that on its own. Ideally this should only be done when looking up a handle provided by the controller, and not everywhere.

Reply:
> `bt_conn_lookup_handle` is a local function to get a local `bt_conn` based on a local handle. By "local" I mean this isn't related to the controller, which is what we seem to want to guard against.

Could you clarify a bit what you mean by the above? `struct bt_conn` is a "local" host-side abstraction for connections, however conn->handle is always the value we get from the controller and what we pass back to the controller to refer to a connection.

Reply:
> however conn->handle is always the value we get from the controller and what we pass back to the controller to refer to a connection.

I think I need to retract my previous comment. I was under the impression that `bt_conn_lookup_handle` was used in other contexts besides getting a `bt_conn` from a controller event, but it seems that I am incorrect; it seems to be exclusively used for that purpose, in which case it makes sense to do it this way. 

_if_ it was used by the host from other host actions, then I think my comment still applies, but I agree that the way that this is used currently, this is a OK way of doing it. 

My bad for not checking properly before commenting :)